### PR TITLE
Disable Buy Button based on availability

### PR DESCRIPTION
**Pull request by israelswf**
## What's the purpose of this pull request?
The purpose of this PR is to `enable / disable` the `Buy Button`, in the PDP page, based on the product's `availability`. 

## How it works? 
When a product is available, the API returns the `https://schema.org/InStock` value. So, basically the `PDP` uses that information to decide whether the `BuyButton` should be enabled or disabled.
image

**Example of a** `OutOfStock` **product**:
image
image

## References
Jira task [here](https://vtex-dev.atlassian.net/browse/FSSS-87?atlOrigin=eyJpIjoiZjAzYTc1MzRjZmVlNDU2NmFmNTJhZDg2NWY5OWExNTIiLCJwIjoiaiJ9).

### CODE REVIEW

**Review by filipewl — APPROVED** [review]
https://schema.org/LGTM! 😂

File: src/components/sections/ProductDetails/ProductDetails.tsx [hunk 1]
```diff
@@ -44,14 +44,15 @@ function ProductDetails({ product: staleProduct }: Props) {
...
-        offers: [{ price, listPrice, seller }],
+        offers: [{ availability, price, listPrice, seller }],
         lowPrice: spotPrice,
       },
       breadcrumbList,
     },
   } = data
 
   const { currency } = useSession()
+  const buyDisabled: boolean = availability !== 'https://schema.org/InStock'
```
Comment (filipewl) [inline src/components/sections/ProductDetails/ProductDetails.tsx:55]: Interesting!
https://github.com/vtex/faststore/blob/653c9d8e54558c19b9ee7aa8c3387d56ed2d16bb/packages/api/src/platforms/vtex/resolvers/offer.ts#L14-L17

**Review by saranicoly — APPROVED** [review]
Nice job!

File: src/components/sections/ProductDetails/ProductDetails.tsx (hunk 1 above)
Comment (eduardoformiga) [reply src/components/sections/ProductDetails/ProductDetails.tsx:55]: I think we don't need the `boolean` type here due to the implicit type.

**Review by eduardoformiga — APPROVED** [review]
LGTM!

**Pull request merged**